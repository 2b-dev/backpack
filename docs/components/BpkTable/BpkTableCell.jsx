import React from 'react'
import CssModules from 'react-css-modules'

import styles from './bpk-table.scss'

const BpkTableCell = (props) => <td styleName='bpk-table__cell' {...props} />

export default CssModules(BpkTableCell, styles)
