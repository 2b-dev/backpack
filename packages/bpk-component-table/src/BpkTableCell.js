import React, { PropTypes } from 'react'

import './bpk-table.scss'

const BpkTableCell = (props) => <td className='bpk-table__cell' children={props.children} />

BpkTableCell.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

export default BpkTableCell
