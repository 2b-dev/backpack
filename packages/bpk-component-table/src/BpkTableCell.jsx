import React, { PropTypes } from 'react'

import './bpk-table.scss'

const BpkTableCell = ({ children }) => <td className='bpk-table__cell' children={children} />

BpkTableCell.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

export default BpkTableCell
