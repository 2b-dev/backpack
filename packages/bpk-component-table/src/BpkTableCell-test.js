import React from 'react';
import renderer from 'react-test-renderer';
import BpkTableCell from './BpkTableCell';

describe('BpkTableCell', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<BpkTableCell>Tabular data</BpkTableCell>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
