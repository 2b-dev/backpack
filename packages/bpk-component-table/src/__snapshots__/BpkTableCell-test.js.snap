// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BpkTableCell should render correctly 1`] = `
<td
  className="bpk-table__cell"
>
  Tabular data
</td>
`;
